library/ssl_tls.c

```diff
@@ -3640,5693 +3360,6043 @@ int mbedtls_ssl_get_session(const mbedtls_ssl_context *ssl,
-
-#if defined(MBEDTLS_X509_CRT_PARSE_C) && \
-    defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
-    /*
-     * Free only the linked list wrapper, not the keys themselves
-     * since the belong to the SNI callback
-     */
-    ssl_key_cert_free(handshake->sni_key_cert);
-#endif /* MBEDTLS_X509_CRT_PARSE_C && MBEDTLS_SSL_SERVER_NAME_INDICATION */
-
-#if defined(MBEDTLS_SSL_ECP_RESTARTABLE_ENABLED)
-    mbedtls_x509_crt_restart_free(&handshake->ecrs_ctx);
-    if (handshake->ecrs_peer_cert != NULL) {
-        mbedtls_x509_crt_free(handshake->ecrs_peer_cert);
-        mbedtls_free(handshake->ecrs_peer_cert);
-    }
-#endif
-
-#if defined(MBEDTLS_X509_CRT_PARSE_C) &&        \
-    !defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
-    mbedtls_pk_free(&handshake->peer_pubkey);
-#endif /* MBEDTLS_X509_CRT_PARSE_C && !MBEDTLS_SSL_KEEP_PEER_CERTIFICATE */
-
-#if defined(MBEDTLS_SSL_CLI_C) && \
-    (defined(MBEDTLS_SSL_PROTO_DTLS) || defined(MBEDTLS_SSL_PROTO_TLS1_3))
-    mbedtls_free(handshake->cookie);
-#endif /* MBEDTLS_SSL_CLI_C &&
-          ( MBEDTLS_SSL_PROTO_DTLS || MBEDTLS_SSL_PROTO_TLS1_3 ) */
-
-#if defined(MBEDTLS_SSL_PROTO_DTLS)
-    mbedtls_ssl_flight_free(handshake->flight);
-    mbedtls_ssl_buffering_free(ssl);
-#endif /* MBEDTLS_SSL_PROTO_DTLS */
-
-#if defined(MBEDTLS_KEY_EXCHANGE_SOME_XXDH_PSA_ANY_ENABLED)
-    if (handshake->xxdh_psa_privkey_is_external == 0) {
-        psa_destroy_key(handshake->xxdh_psa_privkey);
-    }
-#endif /* MBEDTLS_KEY_EXCHANGE_SOME_XXDH_PSA_ANY_ENABLED */
-
-#if defined(MBEDTLS_SSL_PROTO_TLS1_3)
-    mbedtls_ssl_transform_free(handshake->transform_handshake);
-    mbedtls_free(handshake->transform_handshake);
-#if defined(MBEDTLS_SSL_EARLY_DATA)
-    mbedtls_ssl_transform_free(handshake->transform_earlydata);
-    mbedtls_free(handshake->transform_earlydata);
-#endif
-#endif /* MBEDTLS_SSL_PROTO_TLS1_3 */
-
-
-#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
-    /* If the buffers are too big - reallocate. Because of the way Mbed TLS
-     * processes datagrams and the fact that a datagram is allowed to have
-     * several records in it, it is possible that the I/O buffers are not
-     * empty at this stage */
-    handle_buffer_resizing(ssl, 1, mbedtls_ssl_get_input_buflen(ssl),
-                           mbedtls_ssl_get_output_buflen(ssl));
-#endif
-
-    /* mbedtls_platform_zeroize MUST be last one in this function */
-    mbedtls_platform_zeroize(handshake,
-                             sizeof(mbedtls_ssl_handshake_params));
-}
-
-void mbedtls_ssl_session_free(mbedtls_ssl_session *session)
-{
-    if (session == NULL) {
-        return;
-    }
+#define SSL_SERIALIZED_SESSION_CONFIG_CRT 0
+#endif /* MBEDTLS_X509_CRT_PARSE_C */
 
-#if defined(MBEDTLS_X509_CRT_PARSE_C)
-    ssl_clear_peer_cert(session);
-#endif
+#if defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+#define SSL_SERIALIZED_SESSION_KEEP_PEER_CRT 1
+#else
+#define SSL_SERIALIZED_SESSION_KEEP_PEER_CRT 0
```

Comment:
(minor, naming) All the other flags have `CONFIG` in their name, which would make this `SSL_SERIALIZED_SESSION_CONFIG_KEEP_PEER_CRT`